client/src/components/CommentApp/comments.js

```diff
@@ -0,0 +1,144 @@
+import { initCommentsApp } from 'wagtail-comment-frontend';
+import { STRINGS } from '../../config/wagtailConfig';
+
+function initComments(element, author, initialComments) {
+  const app = initCommentsApp(element, author, initialComments, STRINGS);
+  window.commentApp = app;
+  return app;
+}
+
+function getContentPath(fieldNode) {
+  // Return the total contentpath for an element as a string, in the form field.streamfield_uid.block...
+  if (fieldNode.closest('data-contentpath-disabled')) {
+    return '';
+  }
+  let element = fieldNode.closest('[data-contentpath]');
+  const contentPaths = [];
+  while (element !== null) {
+    contentPaths.push(element.dataset.contentpath);
+    element = element.parentElement.closest('[data-contentpath]');
+  }
+  contentPaths.reverse();
+  return contentPaths.join('.');
+}
+
+class BasicFieldLevelAnnotation {
+  constructor(fieldNode, node) {
+    this.node = node;
+    this.fieldNode = fieldNode;
+    this.position = '';
+  }
+  onDelete() {
+    this.node.remove();
+  }
+  onFocus() {
+    this.node.classList.remove('button-secondary');
+    this.node.ariaLabel = STRINGS.UNFOCUS_COMMENT;
+  }
+  onUnfocus() {
+    this.node.classList.add('button-secondary');
+    this.node.ariaLabel = STRINGS.UNFOCUS_COMMENT;
+    // TODO: ensure comment is focused accessibly when this is clicked,
+    // and that screenreader users can return to the annotation point when desired
+  }
+  show() {
+    this.node.classList.remove('u-hidden');
+  }
+  hide() {
+    this.node.classList.add('u-hidden');
+  }
+  setOnClickHandler(handler) {
+    this.node.addEventListener('click', handler);
+  }
+  getDesiredPosition() {
+    return (
+      this.fieldNode.getBoundingClientRect().top +
+      document.documentElement.scrollTop
+    );
+  }
+}
+
+class FieldLevelCommentWidget {
+  constructor(
+    fieldNode,
+    commentAdditionNode,
+    annotationTemplateNode,
+    commentApp
+  ) {
+    const self = this;
```

Comment:
This was for line 74, but yeah, I could probably just replace this by binding the function directly instead rather than doing it inside `addEventListener`.

Reply:
Removed.